js/main.js

```diff
@@ -71,17 +73,39 @@ function applySpeed(value, ignoreDeath = false) {
     return value * getGameSpeed(ignoreDeath) / updateSpeed;
 }
 
-function getPopulation() {
-    let population = 0;
+//Need to review formula + application in updatePopulation()
+function updateHeat() {
+    let danger = getEffectFromOperations(EffectType.Danger);
+    danger += gameData.currentSkill.getEffect(EffectType.Danger);
+    const military = getEffectFromOperations(EffectType.Military);
+
+    if (danger === military) {

```

Comment:
Alle Zahlen in JS sind immer Floats. Floats direkt per === zu vergleichen ist gefährlich, weil zB 0.999999 und 1.0000001 vlt eigentlich dasselbe in deiner Logik sein sollten.
Leg mal ne methode in utils an: (nur so pseudo code)
```
function nearlyEqual(number a, number b, float epsilon = 128 * Number.EPSILON) {
return abs(a - b) <= epsilon;
}
```
Und dann kannst du die Zeile hier ersetzen durch `if (nearlyEqual(danger, military)) {`